### DISCUSSION TITLE
Does script begin by checking if Windows is running ?

### DISCUSSION THREAD

**Discussion by GrimDawn-max** [Lailloken/Exile-UI#396] [Ideas & Feedback]
Trying to run the utility via AHK installed under Crossover (because I no longer have a Windows box and a VM is awful performance wise to even attempt). AHK runs and the L icon appears, however it does not do anything after this. I can't configure screen resolution or anything else. Wondering if there is a log somewhere for troubleshooting to figure out why the configuration page won't come up so I can configure and move onto using the full features (like I used to when I had a Windows box).

**Comment by Lailloken** [comment]
No, there aren't any logs. During startup, the tool does this:
- check if GDI+ is available (if not, there's an error-message and the tool closes)

- create window-groups so the different PoE client-versions can all be addressed via a single "alias"
- wait for any PoE window to exist (this times out after a minute, and the tool closes)

Does the L icon stay for more than a minute?

**Reply by GrimDawn-max** [reply]
"Does the L icon stay for more than a minute?"
No - it disappears after 1 minute. That would correlate with it not finding a POE window. That is probably because it is looking for a Windows process, which won't exist via GPTK (Apple's new tool to run Windows games on a Mac). I do have POE installed under Crossover, which *might* be seen as a Windows process. I'll take a look and see if it runs under that structure and let you know if that helps or not. If not then will need to determine next steps to get it going.  

Edit: Running via Crossover didn't help. The program is failing to locate the POE process.

**Reply by GrimDawn-max** [reply]
Would it be possible to have a version that does not check if PathOfExile process is running ? On a Mac the AHK syntax won't work (I assume it is using WinExist("ahk_exe PathOfExile.exe). I'm not sure if this can be changed to the path the executable is at on a Mac but it would probably be easier not to have the check at all in this case. 

The other issue (at least for now) would likely be the program looking for Client.txt. This exists but not as per a Windows path (it could be in 3 different places depending on how someone runs it). 

For now a test version with no POE check would do. This would allow me to run and see what the next roadblock may potentially be.

**Reply by Lailloken** [reply]
> Would it be possible to have a version that does not check if PathOfExile process is running

Ouch, that would probably set off a huge domino effect. Many hotkeys and functions are reliant on the game-client being active, and/or need to target that window specifically.

You'll probably have to play around with AHK and find out how it handles and/or addresses windows/apps when run via crossover, either by creating test scripts or using the built-in "Window Spy" (usually accessed by right-clicking the AHK tray icon).

**Reply by GrimDawn-max** [reply]
Yeah - I started walking through the main script after I sent that request. There's a lot of calls that also might not work since they are Windows based. I read someone got the program running in Linux. I would expect they ran in to the same issues. The program would not be able to find PathOfExile process either, so wonder how far they actually got or if they just gave up. Not sure if they are still around. Will see if I can ping them. 

For now I think I'm at a dead end. Either I figure out how to make POE playable in a VM again or I go get me a Windows box for this. Theoretically AppleScript could do similar things but I've nowhere near the expertise for that. 

Appreciate you being responsive, as always, to the initial request. I'll put this to bed for now, unless I come up with a break through later on. Thanks !
